### PR TITLE
Context Provider - balance

### LINKED ISSUE #49: Context Provider - balance
Blocking #45

Create a context provider for loading balances. This data should be fetched one time, then used in many places in the app like so:

```js
const { balance } = useBalance()
```

https://www.bundleapps.io/blog/nextjs-context-api-tutorial

### PR DESCRIPTION
**Pull request by alekseevaiana**
closes #49

### CODE REVIEW

File: context/BalanceContext.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      const urlMATIC = `https://api.usekeyp.com/v1/users/${userId}/balance`;
+      const urlNotMATIC = `https://api.usekeyp.com/v1/users/${userId}/balance/${tokenAddress}`;
+      const response = await axios.get(
+        asset === "MATIC" ? urlMATIC : urlNotMATIC,
+        options
+      );
+      const contractAddress = Object.keys(response.data)[0]; // Get the first key from the object
+      const formattedBalance = response.data[contractAddress].formatted;
+      setBalance(Number(formattedBalance).toFixed(4));
+      setLoading(false);
```
Comment (pi0neerpat) [inline context/BalanceContext.tsx:48]: All the balances should be returned

```js
const { balances } = useBalance()

console.log(balances) should output all the return objects from the api

**Review by pi0neerpat — CHANGES_REQUESTED** [review]

**Comment by alekseevaiana** [comment]
Now balances return an object with all available assets

**Review by pi0neerpat — CHANGES_REQUESTED** [review]
Looking great! Only thing left is using the context hook for the asset list as well.

**Comment by alekseevaiana** [comment]
Now we use context provider `const { balances } = useBalance();` in `Balances.tsx` component. Now it fetches when page reloaded and asset changed. It doesn't require refetching when nothing changed. 

https://github.com/UseKeyp/kaching-app/assets/36742189/4f93f7ab-0b97-4f33-b5db-315ac18823a1


https://github.com/UseKeyp/kaching-app/assets/36742189/bd0e54b7-08c0-432f-a355-047bf196f521

**Review by pi0neerpat — APPROVED** [review]
Excellent

**Pull request merged**